Jetsnack/app/src/main/java/com/example/jetsnack/ui/JetsnackApp.kt

```diff
@@ -17,30 +17,65 @@
 package com.example.jetsnack.ui
 
 import androidx.compose.foundation.layout.padding
+import androidx.compose.material.SnackbarHost
+import androidx.compose.material.rememberScaffoldState
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
 import androidx.compose.runtime.remember
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
 import androidx.navigation.compose.rememberNavController
+import com.example.jetsnack.model.SnackbarManager
 import com.example.jetsnack.ui.components.JetsnackScaffold
+import com.example.jetsnack.ui.components.JetsnackSnackbar
 import com.example.jetsnack.ui.home.HomeSections
 import com.example.jetsnack.ui.home.JetsnackBottomBar
 import com.example.jetsnack.ui.theme.JetsnackTheme
 import com.google.accompanist.insets.ProvideWindowInsets
+import com.google.accompanist.insets.systemBarsPadding
 
 @Composable
 fun JetsnackApp() {
     ProvideWindowInsets {
         JetsnackTheme {
             val tabs = remember { HomeSections.values() }
             val navController = rememberNavController()
+            val scaffoldState = rememberScaffoldState()
+
             JetsnackScaffold(
-                bottomBar = { JetsnackBottomBar(navController = navController, tabs = tabs) }
+                bottomBar = { JetsnackBottomBar(navController = navController, tabs = tabs) },
+                snackbarHost = {
+                    SnackbarHost(
+                        hostState = it,
+                        modifier = Modifier.systemBarsPadding(),
+                        snackbar = { snackbarData -> JetsnackSnackbar(snackbarData) }
+                    )
+                },
+                scaffoldState = scaffoldState
             ) { innerPaddingModifier ->
                 JetsnackNavGraph(
                     navController = navController,
                     modifier = Modifier.padding(innerPaddingModifier)
                 )
             }
+
+            // Handle Snackbar messages
+            val currentMessages by SnackbarManager.messages.collectAsState()
+            if (currentMessages.isNotEmpty()) {
+                val message = currentMessages[0]
+                val messageText: String = stringResource(message.messageId)
+
+                // Effect running in a coroutine that displays the Snackbar on the screen
+                // If there's a change to messageText, SnackbarManager, or scaffoldState, the
+                // previous effect will be cancelled and a new one will start with the new values
+                LaunchedEffect(messageText, SnackbarManager, scaffoldState) {
+                    scaffoldState.snackbarHostState.showSnackbar(messageText)
```

Comment:
Can you add a comment to explain that this call blocks (within the coroutine) until the Snackbar gets dismissed (either by the user or by the timeout). This is not really clear when reading this code, so it looks as though the ViewModel is directly notified when the snackbar starts showing.

Reply:
Added more comments, thank you